hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -793,7 +880,18 @@ public static boolean scmBootstrap(OzoneConfiguration conf)
         // SCM Node info containing hostname to scm Id mappings
         // will be persisted into the version file once this node gets added
         // to existing SCM ring post node regular start up.
+
+        if(OzoneSecurityUtil.isSecurityEnabled(conf)) {
+          HASecurityUtils.initializeSecurity(scmStorageConfig,
+              scmInfo.getScmId(), config, getScmAddress(scmhaNodeDetails, conf),
+              false);
+        }
+        scmStorageConfig.setPrimaryScmNodeId(scmInfo.getScmId());
```

Comment:
the scmId here may not be the primaryScmId, not sure if it is OK.

say current cluster is scm1 and scm2, scm2 is the already bootstrapped scm.
now bootstrap scm3, scm3 may talk with scm2 to get the scmInfo, and treate scm2 as the primary SCM.

Reply:
Primary SCM only starts Root CA that is the only one can issue scm sub ca certificates.

But if you add later on, and the leader changes there will be no rootCA running on that SCM, so bootstrap won't be successful. Let me try to handle that.

Reply:
To add more info, the main reason for persistence to all SCM is, so that each SCM knows who is primary and it will help during debugging customer issues. And also in code it is used to start RootCA only on primary SCM when it is matching with SCMID.

Reply:
Updated to fix the issue gracefully.

**The current solution  in PR is** 
1. Removed isLeader check and added a check if it is primary SCM it can issue SCM signed certificate. So, in case leader changes still newly bootstrapped SCM can get signed cert.
2. But if the primary is down, it will fail after failover retry count, instead of after 3 retries to SCM. (Will open a new Jira to address this)

Reply:
Opened https://issues.apache.org/jira/browse/HDDS-5027 for this

Reply:
bq. will be no rootCA running on that SCM, so bootstrap won't be successful

If I remember correctly, the primary SCM(rootCA) does not have to be the same as the Ratis leader. But when primary SCM issue SCM certificate, it will go though via Ratis.

Reply:
Yes that is the plan as discussed, will do that change in HDDS-5027

Reply:
Ok, sounds good to me.